src/scripts/build_nota.py

```diff
@@ -0,0 +1,158 @@
+    """Downloads the Nota dataset.
+
+    Args:
+        destination_dir: The directory to download the dataset to.
+    """
+    destination_dir = Path(destination_dir)
+    destination_dir.mkdir(parents=True, exist_ok=True)
+    outer_pbar_desc = f"Downloading Nota files to the '{destination_dir}' directory"
+    for url_suffix in tqdm(URL_SUFFIXES, desc=outer_pbar_desc):
+        # Define the URL to download the dataset from
+        url = BASE_URL + url_suffix
+
+        # Get URLs to all files on the URL
+        regex = re.compile(r"<a href=\"(.+?)\">")
+        all_files: List[str] = regex.findall(str(requests.get(url).content))
+
+        # Ignore the Parent and README files
+        all_files_filtered = list(
+            filter(lambda x: x != "Readme.txt" and x != "/nota/", all_files)
+        )
+
+        # Remove empty files from the list
+        empty_files = {"INSL20210003.zip"}
+        all_files_filtered = list(set(all_files_filtered) - empty_files)
+
+        # Remove already downloaded files
+        downloaded_files = {
+            f"{path.stem}.zip" for path in Path.cwd().iterdir() if path.is_dir()
+        }
+        all_files_filtered = list(set(all_files_filtered) - downloaded_files)
+
+        # Iterate over the files and download them
+        inner_pbar_desc = f"Downloading files from {url}"
+        for file in tqdm(all_files_filtered, leave=False, desc=inner_pbar_desc):
+            file_url: str = (url + file).replace("%20", " ")
+            while True:
+                try:
+                    destination_path = Path(destination_dir) / file
+                    stream_download(url=file_url, destination_path=destination_path)
+                    break
+                except ContentTooShortError:
+                    sleep(5)
+                    continue
+
+            # Extract the dataset
+            with ZipFile(destination_path, "r") as zip_file:
+                zip_file.extractall(path=destination_dir)
+            Path(destination_path).unlink()
+
+
+def build_huggingface_dataset(dataset_dir: Path | str) -> DatasetDict:
+    """Builds the HuggingFace dataset.
+
+    Args:
+        dataset_dir: The directory to build the dataset from.
+
+    Returns:
+        The HuggingFace dataset.
+    """
+    dataset_dir = Path(dataset_dir)
+
+    # Build metadata file for the dataset
+    metadata_dict: dict[str, list[str]] = dict(audio=[], text=[])
+    for folder in dataset_dir.iterdir():
+        if not folder.is_dir():
+            continue
+        for text_file in folder.glob("*.txt"):
+            wav_filename = text_file.stem + ".wav"
+            metadata_dict["audio"].append(str(folder / wav_filename))
+            metadata_dict["text"].append(text_file.read_text(encoding="utf-8"))
+
+    metadata_df = pd.DataFrame(metadata_dict)
+
+    # Build the dataset
+    dataset = Dataset.from_pandas(metadata_df, preserve_index=False)
+    dataset = dataset.cast_column("audio", Audio(sampling_rate=16000))
```

Comment:
Maybe have sampling rate as a top level constant like BASE URL as this is a specific Nota value

Reply:
Included it now 🙂